--- ./configure.in.orig	2008-10-13 09:49:41.000000000 +0200
+++ ./configure.in	2008-10-15 17:26:10.000000000 +0200
@@ -38,6 +38,10 @@ AM_CONDITIONAL([HAVE_OS_CYGWIN], [test x
 if test x"`uname | grep \"CYGWIN\"`" != x""; then
 	AC_DEFINE([HAVE_OS_CYGWIN], [1], [Compiling under cygwin])
 fi
+AM_CONDITIONAL([HAVE_OS_DARWIN], [test x"`uname`" = x"Darwin"])
+if test x"`uname`" = x"Darwin"; then
+	AC_DEFINE([HAVE_OS_DARWIN], [1], [Compiling under darwin])
+fi
 
 dnl check for basic programs
 AC_PROG_CC()
--- ./configure.orig	2008-10-13 09:50:11.000000000 +0200
+++ ./configure	2008-10-15 17:27:13.000000000 +0200
@@ -857,6 +857,8 @@ GREP
 EGREP
 HAVE_OS_CYGWIN_TRUE
 HAVE_OS_CYGWIN_FALSE
+HAVE_OS_DARWIN_TRUE
+HAVE_OS_DARWIN_FALSE
 INTLTOOL_DESKTOP_RULE
 INTLTOOL_DIRECTORY_RULE
 INTLTOOL_KEYS_RULE
@@ -4599,6 +4601,21 @@ cat >>confdefs.h <<\_ACEOF
 _ACEOF
 
 fi
+ if test x"`uname`" = x"Darwin"; then
+  HAVE_OS_DARWIN_TRUE=
+  HAVE_OS_DARWIN_FALSE='#'
+else
+  HAVE_OS_DARWIN_TRUE='#'
+  HAVE_OS_DARWIN_FALSE=
+fi
+
+if test x"`uname`" = x"Darwin"; then
+
+cat >>confdefs.h <<\_ACEOF
+#define HAVE_OS_DARWIN 1
+_ACEOF
+
+fi
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
@@ -6661,7 +6678,7 @@ ia64-*-hpux*)
   ;;
 *-*-irix6*)
   # Find out which ABI we are using.
-  echo '#line 6664 "configure"' > conftest.$ac_ext
+  echo '#line 6681 "configure"' > conftest.$ac_ext
   if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
   (eval $ac_compile) 2>&5
   ac_status=$?
@@ -9244,11 +9261,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9247: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9264: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:9251: \$? = $ac_status" >&5
+   echo "$as_me:9268: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -9534,11 +9551,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9537: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9554: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:9541: \$? = $ac_status" >&5
+   echo "$as_me:9558: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -9638,11 +9655,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9641: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9658: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:9645: \$? = $ac_status" >&5
+   echo "$as_me:9662: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -12015,7 +12032,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<EOF
-#line 12018 "configure"
+#line 12035 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -12115,7 +12132,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<EOF
-#line 12118 "configure"
+#line 12135 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -14516,11 +14533,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:14519: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:14536: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:14523: \$? = $ac_status" >&5
+   echo "$as_me:14540: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -14620,11 +14637,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:14623: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:14640: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:14627: \$? = $ac_status" >&5
+   echo "$as_me:14644: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -16218,11 +16235,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16221: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16238: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:16225: \$? = $ac_status" >&5
+   echo "$as_me:16242: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -16322,11 +16339,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16325: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16342: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:16329: \$? = $ac_status" >&5
+   echo "$as_me:16346: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -18542,11 +18559,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:18545: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:18562: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:18549: \$? = $ac_status" >&5
+   echo "$as_me:18566: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -18832,11 +18849,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:18835: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:18852: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:18839: \$? = $ac_status" >&5
+   echo "$as_me:18856: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -18936,11 +18953,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:18939: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:18956: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:18943: \$? = $ac_status" >&5
+   echo "$as_me:18960: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -27063,6 +27080,13 @@ echo "$as_me: error: conditional \"HAVE_
 Usually this means the macro was only invoked conditionally." >&2;}
    { (exit 1); exit 1; }; }
 fi
+if test -z "${HAVE_OS_DARWIN_TRUE}" && test -z "${HAVE_OS_DARWIN_FALSE}"; then
+  { { echo "$as_me:$LINENO: error: conditional \"HAVE_OS_DARWIN\" was never defined.
+Usually this means the macro was only invoked conditionally." >&5
+echo "$as_me: error: conditional \"HAVE_OS_DARWIN\" was never defined.
+Usually this means the macro was only invoked conditionally." >&2;}
+   { (exit 1); exit 1; }; }
+fi
 if test -z "${am__fastdepCC_TRUE}" && test -z "${am__fastdepCC_FALSE}"; then
   { { echo "$as_me:$LINENO: error: conditional \"am__fastdepCC\" was never defined.
 Usually this means the macro was only invoked conditionally." >&5
@@ -27745,6 +27769,8 @@ GREP!$GREP$ac_delim
 EGREP!$EGREP$ac_delim
 HAVE_OS_CYGWIN_TRUE!$HAVE_OS_CYGWIN_TRUE$ac_delim
 HAVE_OS_CYGWIN_FALSE!$HAVE_OS_CYGWIN_FALSE$ac_delim
+HAVE_OS_DARWIN_TRUE!$HAVE_OS_DARWIN_TRUE$ac_delim
+HAVE_OS_DARWIN_FALSE!$HAVE_OS_DARWIN_FALSE$ac_delim
 INTLTOOL_DESKTOP_RULE!$INTLTOOL_DESKTOP_RULE$ac_delim
 INTLTOOL_DIRECTORY_RULE!$INTLTOOL_DIRECTORY_RULE$ac_delim
 INTLTOOL_KEYS_RULE!$INTLTOOL_KEYS_RULE$ac_delim
@@ -27757,8 +27783,6 @@ INTLTOOL_SOUNDLIST_RULE!$INTLTOOL_SOUNDL
 INTLTOOL_UI_RULE!$INTLTOOL_UI_RULE$ac_delim
 INTLTOOL_XAM_RULE!$INTLTOOL_XAM_RULE$ac_delim
 INTLTOOL_KBD_RULE!$INTLTOOL_KBD_RULE$ac_delim
-INTLTOOL_XML_RULE!$INTLTOOL_XML_RULE$ac_delim
-INTLTOOL_XML_NOMERGE_RULE!$INTLTOOL_XML_NOMERGE_RULE$ac_delim
 _ACEOF
 
   if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 97; then
@@ -27800,6 +27824,8 @@ _ACEOF
 ac_delim='%!_!# '
 for ac_last_try in false false false false false :; do
   cat >conf$$subs.sed <<_ACEOF
+INTLTOOL_XML_RULE!$INTLTOOL_XML_RULE$ac_delim
+INTLTOOL_XML_NOMERGE_RULE!$INTLTOOL_XML_NOMERGE_RULE$ac_delim
 INTLTOOL_CAVES_RULE!$INTLTOOL_CAVES_RULE$ac_delim
 INTLTOOL_SCHEMAS_RULE!$INTLTOOL_SCHEMAS_RULE$ac_delim
 INTLTOOL_THEME_RULE!$INTLTOOL_THEME_RULE$ac_delim
@@ -27895,8 +27921,6 @@ LIBGLADE_VERSION!$LIBGLADE_VERSION$ac_de
 LIBGLADE_CFLAGS!$LIBGLADE_CFLAGS$ac_delim
 LIBGLADE_LIBS!$LIBGLADE_LIBS$ac_delim
 LIBGLADE_REQUIRED_VERSION!$LIBGLADE_REQUIRED_VERSION$ac_delim
-GNOME_VERSION!$GNOME_VERSION$ac_delim
-GNOME_CFLAGS!$GNOME_CFLAGS$ac_delim
 _ACEOF
 
   if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 97; then
@@ -27938,6 +27962,8 @@ _ACEOF
 ac_delim='%!_!# '
 for ac_last_try in false false false false false :; do
   cat >conf$$subs.sed <<_ACEOF
+GNOME_VERSION!$GNOME_VERSION$ac_delim
+GNOME_CFLAGS!$GNOME_CFLAGS$ac_delim
 GNOME_LIBS!$GNOME_LIBS$ac_delim
 GNOME_REQUIRED_VERSION!$GNOME_REQUIRED_VERSION$ac_delim
 HAVE_GNOME_TRUE!$HAVE_GNOME_TRUE$ac_delim
@@ -27954,7 +27980,7 @@ LIBOBJS!$LIBOBJS$ac_delim
 LTLIBOBJS!$LTLIBOBJS$ac_delim
 _ACEOF
 
-  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 14; then
+  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 16; then
     break
   elif $ac_last_try; then
     { { echo "$as_me:$LINENO: error: could not make $CONFIG_STATUS" >&5
--- ./libxfsm/Makefile.am.orig	2008-10-11 14:25:15.000000000 +0200
+++ ./libxfsm/Makefile.am	2008-10-15 17:26:10.000000000 +0200
@@ -30,6 +30,12 @@ libxfsm_4_6_la_LIBADD =							\
 	$(XFCONF_LIBS)
 endif
 
+if HAVE_OS_DARWIN
+libxfsm_4_6_la_LIBADD =							\
+	$(LIBXFCEGUI4_LIBS)						\
+	$(XFCONF_LIBS)
+endif
+
 
 libxfsmincludedir =							\
 	$(includedir)/xfce4/xfce4-session-4.6/libxfsm
--- ./libxfsm/Makefile.in.orig	2008-10-13 09:50:07.000000000 +0200
+++ ./libxfsm/Makefile.in	2008-10-15 17:26:11.000000000 +0200
@@ -316,6 +316,10 @@
 @HAVE_OS_CYGWIN_TRUE@	$(LIBXFCEGUI4_LIBS)						\
 @HAVE_OS_CYGWIN_TRUE@	$(XFCONF_LIBS)
 
+@HAVE_OS_DARWIN_TRUE@libxfsm_4_6_la_LIBADD = \
+@HAVE_OS_DARWIN_TRUE@	$(LIBXFCEGUI4_LIBS)						\
+@HAVE_OS_DARWIN_TRUE@	$(XFCONF_LIBS)
+
 libxfsmincludedir = \
 	$(includedir)/xfce4/xfce4-session-4.6/libxfsm
 
