--- ./configure.in.orig	2008-09-11 00:25:38.000000000 +0200
+++ ./configure.in	2008-09-30 22:21:48.000000000 +0200
@@ -38,6 +38,10 @@ AM_CONDITIONAL([HAVE_OS_CYGWIN], [test x
 if test x"`uname | grep \"CYGWIN\"`" != x""; then
 	AC_DEFINE([HAVE_OS_CYGWIN], [1], [Compiling under cygwin])
 fi
+AM_CONDITIONAL([HAVE_OS_DARWIN], [test x"`uname | grep \"DARWIN\"`" != x""])
+if test "`uname`" = "Darwin"; then
+	AC_DEFINE([HAVE_OS_DARWIN], [1], [Compiling under darwin])
+fi
 
 dnl check for basic programs
 AC_PROG_CC()
--- ./configure.orig	2008-09-11 00:26:31.000000000 +0200
+++ ./configure	2008-09-30 22:22:21.000000000 +0200
@@ -857,6 +857,8 @@ GREP
 EGREP
 HAVE_OS_CYGWIN_TRUE
 HAVE_OS_CYGWIN_FALSE
+HAVE_OS_DARWIN_TRUE
+HAVE_OS_DARWIN_FALSE
 INTLTOOL_DESKTOP_RULE
 INTLTOOL_DIRECTORY_RULE
 INTLTOOL_KEYS_RULE
@@ -4580,6 +4582,21 @@ cat >>confdefs.h <<\_ACEOF
 _ACEOF
 
 fi
+ if test "`uname`" = "Darwin"; then
+  HAVE_OS_DARWIN_TRUE=
+  HAVE_OS_DARWIN_FALSE='#'
+else
+  HAVE_OS_DARWIN_TRUE='#'
+  HAVE_OS_DARWIN_FALSE=
+fi
+
+if test "`uname`" = "Darwin"; then
+
+cat >>confdefs.h <<\_ACEOF
+#define HAVE_OS_DARWIN 1
+_ACEOF
+
+fi
 
 ac_ext=c
 ac_cpp='$CPP $CPPFLAGS'
@@ -6642,7 +6659,7 @@ ia64-*-hpux*)
   ;;
 *-*-irix6*)
   # Find out which ABI we are using.
-  echo '#line 6645 "configure"' > conftest.$ac_ext
+  echo '#line 6662 "configure"' > conftest.$ac_ext
   if { (eval echo "$as_me:$LINENO: \"$ac_compile\"") >&5
   (eval $ac_compile) 2>&5
   ac_status=$?
@@ -9225,11 +9242,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9228: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9245: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:9232: \$? = $ac_status" >&5
+   echo "$as_me:9249: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -9515,11 +9532,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9518: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9535: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:9522: \$? = $ac_status" >&5
+   echo "$as_me:9539: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -9619,11 +9636,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:9622: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:9639: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:9626: \$? = $ac_status" >&5
+   echo "$as_me:9643: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -11996,7 +12013,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<EOF
-#line 11999 "configure"
+#line 12016 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -12096,7 +12113,7 @@ else
   lt_dlunknown=0; lt_dlno_uscore=1; lt_dlneed_uscore=2
   lt_status=$lt_dlunknown
   cat > conftest.$ac_ext <<EOF
-#line 12099 "configure"
+#line 12116 "configure"
 #include "confdefs.h"
 
 #if HAVE_DLFCN_H
@@ -14497,11 +14514,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:14500: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:14517: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:14504: \$? = $ac_status" >&5
+   echo "$as_me:14521: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -14601,11 +14618,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:14604: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:14621: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:14608: \$? = $ac_status" >&5
+   echo "$as_me:14625: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -16199,11 +16216,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16202: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16219: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:16206: \$? = $ac_status" >&5
+   echo "$as_me:16223: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -16303,11 +16320,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:16306: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:16323: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:16310: \$? = $ac_status" >&5
+   echo "$as_me:16327: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -18523,11 +18540,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:18526: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:18543: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:18530: \$? = $ac_status" >&5
+   echo "$as_me:18547: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -18813,11 +18830,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:18816: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:18833: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>conftest.err)
    ac_status=$?
    cat conftest.err >&5
-   echo "$as_me:18820: \$? = $ac_status" >&5
+   echo "$as_me:18837: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s "$ac_outfile"; then
      # The compiler can only warn and ignore the option if not recognized
      # So say no if there are warnings other than the usual output.
@@ -18917,11 +18934,11 @@ else
    -e 's:.*FLAGS}\{0,1\} :&$lt_compiler_flag :; t' \
    -e 's: [^ ]*conftest\.: $lt_compiler_flag&:; t' \
    -e 's:$: $lt_compiler_flag:'`
-   (eval echo "\"\$as_me:18920: $lt_compile\"" >&5)
+   (eval echo "\"\$as_me:18937: $lt_compile\"" >&5)
    (eval "$lt_compile" 2>out/conftest.err)
    ac_status=$?
    cat out/conftest.err >&5
-   echo "$as_me:18924: \$? = $ac_status" >&5
+   echo "$as_me:18941: \$? = $ac_status" >&5
    if (exit $ac_status) && test -s out/conftest2.$ac_objext
    then
      # The compiler can only warn and ignore the option if not recognized
@@ -26489,6 +26506,13 @@ echo "$as_me: error: conditional \"HAVE_
 Usually this means the macro was only invoked conditionally." >&2;}
    { (exit 1); exit 1; }; }
 fi
+if test -z "${HAVE_OS_DARWIN_TRUE}" && test -z "${HAVE_OS_DARWIN_FALSE}"; then
+  { { echo "$as_me:$LINENO: error: conditional \"HAVE_OS_DARWIN\" was never defined.
+Usually this means the macro was only invoked conditionally." >&5
+echo "$as_me: error: conditional \"HAVE_OS_DARWIN\" was never defined.
+Usually this means the macro was only invoked conditionally." >&2;}
+   { (exit 1); exit 1; }; }
+fi
 if test -z "${am__fastdepCC_TRUE}" && test -z "${am__fastdepCC_FALSE}"; then
   { { echo "$as_me:$LINENO: error: conditional \"am__fastdepCC\" was never defined.
 Usually this means the macro was only invoked conditionally." >&5
@@ -27174,6 +27198,8 @@ GREP!$GREP$ac_delim
 EGREP!$EGREP$ac_delim
 HAVE_OS_CYGWIN_TRUE!$HAVE_OS_CYGWIN_TRUE$ac_delim
 HAVE_OS_CYGWIN_FALSE!$HAVE_OS_CYGWIN_FALSE$ac_delim
+HAVE_OS_DARWIN_TRUE!$HAVE_OS_DARWIN_TRUE$ac_delim
+HAVE_OS_DARWIN_FALSE!$HAVE_OS_DARWIN_FALSE$ac_delim
 INTLTOOL_DESKTOP_RULE!$INTLTOOL_DESKTOP_RULE$ac_delim
 INTLTOOL_DIRECTORY_RULE!$INTLTOOL_DIRECTORY_RULE$ac_delim
 INTLTOOL_KEYS_RULE!$INTLTOOL_KEYS_RULE$ac_delim
@@ -27186,8 +27212,6 @@ INTLTOOL_SOUNDLIST_RULE!$INTLTOOL_SOUNDL
 INTLTOOL_UI_RULE!$INTLTOOL_UI_RULE$ac_delim
 INTLTOOL_XAM_RULE!$INTLTOOL_XAM_RULE$ac_delim
 INTLTOOL_KBD_RULE!$INTLTOOL_KBD_RULE$ac_delim
-INTLTOOL_XML_RULE!$INTLTOOL_XML_RULE$ac_delim
-INTLTOOL_XML_NOMERGE_RULE!$INTLTOOL_XML_NOMERGE_RULE$ac_delim
 _ACEOF
 
   if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 97; then
@@ -27229,6 +27253,8 @@ _ACEOF
 ac_delim='%!_!# '
 for ac_last_try in false false false false false :; do
   cat >conf$$subs.sed <<_ACEOF
+INTLTOOL_XML_RULE!$INTLTOOL_XML_RULE$ac_delim
+INTLTOOL_XML_NOMERGE_RULE!$INTLTOOL_XML_NOMERGE_RULE$ac_delim
 INTLTOOL_CAVES_RULE!$INTLTOOL_CAVES_RULE$ac_delim
 INTLTOOL_SCHEMAS_RULE!$INTLTOOL_SCHEMAS_RULE$ac_delim
 INTLTOOL_THEME_RULE!$INTLTOOL_THEME_RULE$ac_delim
@@ -27326,7 +27352,7 @@ LIBOBJS!$LIBOBJS$ac_delim
 LTLIBOBJS!$LTLIBOBJS$ac_delim
 _ACEOF
 
-  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 95; then
+  if test `sed -n "s/.*$ac_delim\$/X/p" conf$$subs.sed | grep -c X` = 97; then
     break
   elif $ac_last_try; then
     { { echo "$as_me:$LINENO: error: could not make $CONFIG_STATUS" >&5
@@ -27345,7 +27371,7 @@ fi
 
 cat >>$CONFIG_STATUS <<_ACEOF
 cat >"\$tmp/subs-2.sed" <<\CEOF$ac_eof
-/@[a-zA-Z_][a-zA-Z_0-9]*@/!b end
+/@[a-zA-Z_][a-zA-Z_0-9]*@/!b
 _ACEOF
 sed '
 s/[,\\&]/\\&/g; s/@/@|#_!!_#|/g
@@ -27358,8 +27384,6 @@ N; s/^.*\n//; s/[,\\&]/\\&/g; s/@/@|#_!!
 ' >>$CONFIG_STATUS <conf$$subs.sed
 rm -f conf$$subs.sed
 cat >>$CONFIG_STATUS <<_ACEOF
-:end
-s/|#_!!_#|//g
 CEOF$ac_eof
 _ACEOF
 
@@ -27613,7 +27637,7 @@ s&@abs_top_builddir@&$ac_abs_top_builddi
 s&@INSTALL@&$ac_INSTALL&;t t
 s&@MKDIR_P@&$ac_MKDIR_P&;t t
 $ac_datarootdir_hack
-" $ac_file_inputs | sed -f "$tmp/subs-1.sed" | sed -f "$tmp/subs-2.sed" >$tmp/out
+" $ac_file_inputs | sed -f "$tmp/subs-1.sed" | sed -f "$tmp/subs-2.sed" | sed 's/|#_!!_#|//g' >$tmp/out
 
 test -z "$ac_datarootdir_hack$ac_datarootdir_seen" &&
   { ac_out=`sed -n '/\${datarootdir}/p' "$tmp/out"`; test -n "$ac_out"; } &&
--- ./libxfsm/Makefile.am.orig	2008-09-11 00:25:25.000000000 +0200
+++ ./libxfsm/Makefile.am	2008-09-30 22:25:02.000000000 +0200
@@ -28,6 +28,11 @@ libxfsm_4_2_la_LIBADD =							\
 	@LIBXFCEGUI4_LIBS@
 endif
 
+if HAVE_OS_DARWIN
+libxfsm_4_2_la_LIBADD =							\
+	@LIBXFCEGUI4_LIBS@
+endif
+
 
 libxfsmincludedir =							\
 	$(includedir)/xfce4/xfce4-session-4.2/libxfsm
--- ./libxfsm/Makefile.in.orig	2008-09-11 00:26:21.000000000 +0200
+++ ./libxfsm/Makefile.in	2008-09-30 22:29:48.000000000 +0200
@@ -294,6 +294,9 @@ libxfsm_4_2_la_LDFLAGS = -export-dynamic
 @HAVE_OS_CYGWIN_TRUE@	@LIBX11_LIBS@							\
 @HAVE_OS_CYGWIN_TRUE@	@LIBXFCEGUI4_LIBS@
 
+@HAVE_OS_DARWIN_TRUE@libxfsm_4_2_la_LIBADD = \
+@HAVE_OS_DARWIN_TRUE@	@LIBXFCEGUI4_LIBS@
+
 libxfsmincludedir = \
 	$(includedir)/xfce4/xfce4-session-4.2/libxfsm
 
